appserver/pom.xml

```diff
@@ -158,7 +158,7 @@
 
         <!-- CDI -->
         <cdi-api.version>3.0.0</cdi-api.version>
-        <weld.version>4.0.0.Alpha3</weld.version>
+        <weld.version>4.0.0.Beta5</weld.version>
         <jboss.classfilewriter.version>1.2.1.Final</jboss.classfilewriter.version>
```

Comment:
there's an update to jboss classfilewriter too, would be nice to include it. Note that #23242 takes care of weld too.